json-core/src/main/java/io/micronaut/json/convert/JsonConverterRegistrar.java

```diff
@@ -211,8 +211,11 @@ private Object correctKeys(Object o) {
                     return Optional.of(new String(objectCodec.get().writeValueAsBytes(node), StandardCharsets.UTF_8));
                 } else {
                     Argument<?> argument = null;
-                    if (context instanceof ArgumentConversionContext && targetType.getTypeParameters().length != 0) {
+                    if (context instanceof ArgumentConversionContext) {
```

Comment:
Do you know what the purpose of the check for no type parameters was? Why did it need to be removed?

Reply:
I believe it was a performance optimization since there was no need to use the `Argument` if there was no generic type information and `targetType` could just be used

Reply:
This is the check that failed the first CI run, see the comment above. There seem to be cases where the argument in the context and the targetType diverge, similar to the response body argument problem graeme had